turbo.json

```
@@ -4,84 +4,74 @@
         "build": {
             "dependsOn": ["^build"],
             "inputs": [
-                "**/tsconfig.json",
-                "packages/**/*.{js,json,ts,sol}",
-                "!packages/**/{artifacts,cache,dist,typechain-types}/**"
+                "tsconfig.json",
+                "**/build.tsconfig.json",
+                "**/rollup.config.ts",
+                "packages/**/*.{js,json,ts}",
+                "!packages/**/{dist}/**"
             ],
-            "outputs": ["packages/**/{artifacts,dist,typechain-types}/**"]
+            "outputs": ["packages/**/{dist}/**"]
         },
-        "//#_build:libraries": {
+        "//#_build": {
             "inputs": [
                 "tsconfig.json",
+                "**/build.tsconfig.json",
+                "**/rollup.config.ts",
                 "packages/**/*.{js,json,ts}",
-                "!packages/**/{artifacts,cache,dist,typechain-types}/**"
+                "!packages/**/{dist}/**"
             ]
         },
+        "//#_test": {
+            "dependsOn": [
+                "@zk-kit/imt#build",
+                "@zk-kit/lean-imt#build",
+                "@zk-kit/baby-jubjub#build",
+                "@zk-kit/eddsa-poseidon#build",
+                "@zk-kit/groth16#build",
+                "@zk-kit/lazytower#build",
+                "@zk-kit/poseidon-cipher#build",
+                "@zk-kit/poseidon-proof#build",
+                "@zk-kit/rollup-plugin-rust#build",
+                "@zk-kit/smt#build",
+                "@zk-kit/utils#build"
+            ],
+            "inputs": ["jest.config.ts", "packages/**/*.{js,ts}", "!packages/**/{dist}/**"]
+        },
         "//#format:prettier": {
             "inputs": [
                 ".prettierrc.json",
                 ".prettierignore",
-                "*.{js,ts,yml,json,md}",
-                "{.github,packages,scripts,types}/**/*.{js,ts,yml,json,md}",
-                "!packages/**/{artifacts,cache,dist,typechain-types}/**"
+                "*.{js,ts,yml,yaml,json,md}",
+                "{.github,packages,scripts,types}/**/*.{js,ts,yml,yaml,json,md}",
+                "!packages/**/{dist,node_modules}/**"
```

```suggestion
                "!packages/**/dist/**"
```

Didn't test, but my understanding of the turbo docs is that [everything that is gitignored is already excluded from the inputs](https://turbo.build/repo/docs/core-concepts/caching/file-inputs#turborepos-default-behavior). So no need to add ignore `node_modules` here.

---

Ohh, so we don't need `dist` either here no?

---

hmm you're right ^^